billing/credit/service.go

```diff
@@ -80,6 +80,16 @@ func (s Service) Deduct(ctx context.Context, cred Credit) error {
 		return errors.New("credit amount is negative")
 	}
 
+	//check if already deducted
+	t, err := s.transactionRepository.GetByID(ctx, cred.ID)
```

Comment:
Instead of adding a DB call, can we try to handle the DB error itself, and return a 409 in that case?
That is, in the repository, when we see a failure, check to see if it is due to "already deduced transaction", and if so, bubble up an appropriate error.

Reply:
Updated the code to handle the duplicate error at repository level and propagate errors upwards.